middleware/response_writer.go

```diff
@@ -150,6 +163,8 @@ func (m ResponseWriter) writeResponse(w http.ResponseWriter, r *http.Request, re
 			return errMarshal
 		}
 
+		w.WriteHeader(m.StatusCode)
+
```

Comment:
@Dmouri Not really obvious, but calling `WriteHeader` after setting content type fix up the issue of not getting the proper content type value.